### ISSUE TITLE
Throw exception when running a dynamic LCM problem with no density specified

### ISSUE THREAD

**Issue by ikalash** [gahansen/Albany#284]
The default value of the density in the code is 1.0 which is non-physical.  Exception should be thrown when not specifying density for dynamic problem to prevent user from running with density = 1.0 accidentally.

**Comment by djlittl** [comment]
Agreed!  A few years ago I spent several hours debugging a simulation, only to find that I had a typo in the input deck and Albany/LCM was using a default value for the elastic modulus.  You guessed it, the default was 1.0, only off by ~9 orders of magnitude ...

**Comment by ikalash** [comment]
@lxmota (others welcome to join in) : should we disallow default values for the material properties for all LCM problems or only dynamic ones?  I hope to finally have time to fix this issue.

**Comment by jwfoulk** [comment]
I guess I'm tempted to disallow defaults in parameters that are requisite for solving the problem. In this sense, you don't need density for quasi-statics but it is needed for dynamics. Clearly, some constitutive model is needed in both cases so having default values for things like the modulus is probably not a good thing.

**Comment by ikalash** [comment]
This is what I thought based on our discussion yesterday, thanks for confirming.  I can make this change.  I anticipate having to modify a lot of input files that _do_ make use of default values (whether intentionally or not).

**Comment by ikalash** [comment]
@jwfoulk : would you be able to give me a list of parameters that are requisite for solving LCM problems?  Some are probably physics specific but I'd guess there are a lot required by all LCM problems.

**Comment by lxmota** [comment]
The parameters unfortunately are material dependent.

If you are using an isotropic elastic model, then one only needs E, nu, and for dynamics, rho.

But there are more complex material models, with many more parameters.

I guess the ConstitutiveModelInterface is where one would disable default values.

**Comment by jwfoulk** [comment]
Yes. Great point. This is what Alejandro is getting at - defaulting any material property can lead to ruin.
